### PR TITLE
move mongoose to a dev dependency. drop buffer support for now

### LINKED ISSUE #6: Installs mongoose as a child dependency
I just noticed that resource-schema installs mongoose as a child dependency in garbanzo:

> kerberos@0.0.7 install /Users/alon/Projects/garbanzo/node_modules/resource-schema/node_modules/mongoose/node_modules/mongodb/node_modules/kerberos

I don't think we want this. It needs to be using the app-level configured instance of mongoose in order to have access to the app's models and the connection. I assume it is written to use the right instance which means this dependency is unnecessary.

Solutions include requiring it as a dev-dependency in the module so that the requiring app has to install mongoose itself. npm dedup could also be a solution. I prefer a solution handled by resource-schema itself rather than being a requirement on the requiring app.

### PR DESCRIPTION
**Pull request by makebbekus**
resolves #6

### CODE REVIEW

File: src/index.coffee [hunk 1]
```diff
@@ -2,7 +2,7 @@ dot = require 'dot-component'
 _ = require 'underscore'
 q = require 'q'
 deepExtend = require './deep_extend'
-mongoose = require 'mongoose'
+ObjectId = require('bson').ObjectID
```
Comment (makebbekus) [inline src/index.coffee]: This is the underlying library mongoose uses.

File: src/index.coffee [hunk 2]
```diff
@@ -500,11 +500,10 @@ module.exports = class ResourceSchema
     for schemaKey in schemaKeys
       instance = Model.schema.paths[schemaKey].instance
       type = switch instance
-        when 'Buffer' then mongoose.Types.Buffer
```
Comment (makebbekus) [inline src/index.coffee]: It doesn't look there is any special handling for Buffer now, so I took it out.

File: package.json [hunk 3]
```diff
@@ -37,7 +38,7 @@
     "cookie-parser": "^1.3.3",
     "body-parser": "^1.9.0",
     "express": "^4.9.7",
-    "request": "^2.45.0",
+    "request": "^2.51.0",
```
Comment (makebbekus) [inline package.json]: I couldn't run npm test with older versions of request.

File: src/index.coffee (hunk 2 above)
Comment (benbuckman) [inline src/index.coffee]: The only reason we'd want a buffer is to store binary data, which we never do.

**Comment by benbuckman** [comment]
:+1:

**Pull request merged**